pkg/backends/tcp.go

```diff
@@ -227,8 +247,9 @@ func (ns *TCPSession) Close() error {
 // Command line
 // **************************************************************************
 
-// tcpListenerCfg is the cmdline configuration object for a TCP listener.
-type tcpListenerCfg struct {
+//TODO make these fields private
```

Comment:
Leaving this as TODO for now as this PR is already 200+ lines of code.

Will make a separate PR refactoring the scope of our fields